ui/src/text_annotator.tsx

```diff
@@ -130,31 +131,33 @@ export
       }
     </div>
   ),
-  XTextAnnotator = ({ model }: { model: TextAnnotator }) => {
+  XTextAnnotator = ({ model: { name, title, tags, items, trigger, readonly, smart_selection = true } }: { model: TextAnnotator }) => {
     const
-      [startIdx, setStartIdx] = React.useState<U | null>(null),
-      [activeTag, setActiveTag] = React.useState<S | undefined>(model.readonly ? undefined : model.tags[0]?.name),
+      [activeTag, setActiveTag] = React.useState<S | undefined>(readonly ? undefined : tags[0]?.name),
       [hoveredTagIdx, setHoveredTagIdx] = React.useState<U | null>(),
-      tagColorMap = model.tags.reduce((map, t) => {
+      tagColorMap = tags.reduce((map, t) => {
```

Comment:
Let's use map instead of reduce: `tagColorMap = new Map(tags.map(t => [t.name, t.color]))` - more elegant.